doc/model.rst

```diff
@@ -0,0 +1,2354 @@
+---------------
+Model functions
+---------------
+A ``Model`` object supports two prefixing functions: ``Model.enable_key_prefixing()`` and
+``Model.register_incoming_entity()``.
+
+To enable prefixing on a ``Model``, users must use the ``Model.enable_key_prefixing()``
+function in the ``Experiment`` driver script. The key components of this function include:
+
+- Activates prefixing for tensors, ``Datasets``, and lists sent to a ``Orchestrator`` from within
+  the ``Model`` application.
+- Enables access to prefixing ``Client`` functions within the ``Model`` application. This excludes
+  the ``Client.set_data_source()`` function, where ``enable_key_prefixing()`` is not require access.
+
+.. note::
+    ML model and script prefixing is not automatically enabled through ``Model.enable_key_prefixing()``
+    and rather must be enabled within the ``Model`` application using ``Client.use_model_ensemble_prefix()``.
+
+To enable a SmartRedis ``Client`` to search a number of ``Model`` prefixes when requesting data from the ``Orchestrator``,
+execute ``Model.register_incoming_entity()`` on the designated ``Model``. The key components of this function include:
+
+- If a ``Model`` application requests prefixed tensors placed by another ``Model``, the interaction
+  between the two entities must be registered in the ``Experiment`` driver script. If the interaction is not
+  properly registered, ``Client.set_data_source()`` will not be able to find the ``Models`` prefix
+  that placed the tensors in the ``Orchestrator``.
+- This function enables users to execute ``Client`` functions such as ``Client.delete_tensor(tensor_name)`` on
+  prefixed tensors without having to specify the prefix. The prefix is instead set via ``Client.set_data_source(model_name)``
+  in the application and registered via ``Model.register_incoming_entity()`` in the ``Experiment`` driver script.
+
+.. _client_prefix_func:
+----------------
+Client functions
+----------------
+A ``Client`` object supports five prefixing functions: ``Client.use_tensor_ensemble_prefix()``,
+``Client.use_dataset_ensemble_prefix()``,  ``Client.use_list_ensemble_prefix()``,
+``Client.use_model_ensemble_prefix()`` and ``Client.set_data_source()``.
+
+To enable or disable SmartRedis data structure prefixing for tensors, ``Datasets``, aggregation lists, ML models
+and scripts, SmartRedis ``Client`` offers functions per data structure:
+
+- Tensor: ``Client.use_tensor_ensemble_prefix()``
+- ``Dataset``: ``Client.use_dataset_ensemble_prefix()``
+- Aggregation lists: ``Client.use_list_ensemble_prefix()``
+- ML models/scripts: ``Client.use_model_ensemble_prefix()``
+
+.. warning::
+    To access the ``Client`` prefixing functions, prefixing must be enabled on the
+    ``Model`` through ``Model.enable_key_prefixing()``. This function activates prefixing
+    for tensors, ``Datasets`` and lists.
+
+For examples on instructing SmartSim to send prefixed data to the ``Orchestrator``, navigate to the
```

Comment:
These functions affect more than just put and get (they affect run as well).  I think this sentence is not needed or should just be generalized to:

Examples are provided below that show the use of these client methods in conjunction with the SmartSim key prefixing ``Model`` API functions.